### ISSUE TITLE
Allow to modify the system prompt after the playground interactions started

### ISSUE THREAD

**Issue by slemeur** [projectatomic/ai-studio#618]
After #617, the user will not be able to modify the system prompt, once the playground interactions started.

In this issue, we would to better handle the situation and let the user modify it, but also understand the impact of that change. 
In order to allow the comparison of the behavior of the model with a different system prompt, the user should still be able to see the previous interactions (and eventually continue testing with more interactions, with the different system prompts). 

**_Proposal:_**
When the user changes the system prompt after having started interacting, we should bring the user into a new playground environment with the new system prompt.

We would display a message asking the user what the desired "re-play" stated is.
Two options to be proposed to the end-user: 
- Does the user wants to replay all the messages sent? 
- Or does the user prefer to start from scratch?

**Comment by axel7083** [comment]
I would argue that re-playing all the conversation does not really make sense, I would propose that only the first message of the user would be replay, as the responses following it depends on the conversation, which do not have the same history after changing the system prompt.

**Comment by feloy** [comment]
> I would argue that re-playing all the conversation does not really make sense, I would propose that only the first message of the user would be replay, as the responses following it depends on the conversation, which do not have the same history after changing the system prompt.

@slemeur wdyt?

**Comment by axel7083** [comment]
> > I would argue that re-playing all the conversation does not really make sense, I would propose that only the first message of the user would be replay, as the responses following it depends on the conversation, which do not have the same history after changing the system prompt.
> 
> @slemeur wdyt?
> 
> In the new playground, it would be nice to have the user proposed the prompt from the previous session, until the user changes the prompt.

I like this idea!

**Comment by slemeur** [comment]
In this flow, we need to let the user deciding how to re-play (or not) the conversation, but the user needs to be aware of what's going to happen, otherwise the experience will be confusing. 

I can imagine the proposed experience to be tedious and long, if the number of prompt is high.

When the user edits the system prompt, we should provide the ability to either:
- Restart a new playground with a different system prompt: in this case the conversation will restart from scratch
- Replay the conversation: in this case, we ask the user the number of messages to be replayed starting from the first one that has been sent. In another issue, we would add the ability to delete a previously sent prompt, which would grant the ability to come back to an earlier point in the conversation.

**Comment by axel7083** [comment]
> In this flow, we need to let the user deciding how to re-play (or not) the conversation, but the user needs to be aware of what's going to happen, otherwise the experience will be confusing.
> 
> I can imagine the proposed experience to be tedious and long, if the number of prompt is high.
> 
> When the user edits the system prompt, we should provide the ability to either:
> 
> * Restart a new playground with a different system prompt: in this case the conversation will restart from scratch

I would agree with this one

> * Replay the conversation: in this case, we ask the user the number of messages to be replayed starting from the first one that has been sent. In another issue, we would add the ability to delete a previously sent prompt, which would grant the ability to come back to an earlier point in the conversation.

This is a very complicated mechanic which would be hard to implement in terms of UI design. I would suggest waiting for user's feedback? To see if we really need such feature, as we have a lot of area of improvement already including the playground base UI

**Comment by feloy** [comment]
> > In this flow, we need to let the user deciding how to re-play (or not) the conversation, but the user needs to be aware of what's going to happen, otherwise the experience will be confusing.
> > I can imagine the proposed experience to be tedious and long, if the number of prompt is high.
> > When the user edits the system prompt, we should provide the ability to either:
> > 
> > * Restart a new playground with a different system prompt: in this case the conversation will restart from scratch
> 
> I would agree with this one
> 

I'm not sure to understand. What would be the difference with the existing possibility to start a new playground?

**Comment by nichjones1** [comment]
might be fixed by other study on new UI for playground environment
